### PR TITLE
Add notes and questions for n20721

### PR DESCRIPTION
**Pull request by jnewbery**

### CODE REVIEW

**Comment by jnewbery** [comment]
I'll push this in an hour or so. @jonatack any review before or after that is very welcome!

**Comment by jonatack** [comment]
Great! Having a look in 15.

File: _posts/2021-02-03-#20721.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    processing is `PeerManager`, which is declared in
+    [net_processing.h](https://github.com/bitcoin/bitcoin/blob/16b784d9/src/net_processing.h).
+    Net processing uses function calls like `ProcessNewBlock()` and
+    `AcceptToMemoryPool()` to call into ...
+
+  - **Validation** - is responsible for our view of the block chain and the UTXO
+    set. It may (or may not) include a mempool for unconfirmed transactions. There
+    are various classes and functions exposed to the rest of the node. [PR
+    20158](https://github.com/bitcoin/bitcoin/pull/20158) and linked PRs are an
+    attempt to clarify and rationlize the interface to validation.
```
Comment (jonatack) [inline _posts/2021-02-03-#20721.md:36]: rationalize

File: _posts/2021-02-03-#20721.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    handles application-level view of peers. The main class in net
+    processing is `PeerManager`, which is declared in
+    [net_processing.h](https://github.com/bitcoin/bitcoin/blob/16b784d9/src/net_processing.h).
+    Net processing uses function calls like `ProcessNewBlock()` and
+    `AcceptToMemoryPool()` to call into ...
+
+  - **Validation** - is responsible for our view of the block chain and the UTXO
+    set. It may (or may not) include a mempool for unconfirmed transactions. There
+    are various classes and functions exposed to the rest of the node. [PR
+    20158](https://github.com/bitcoin/bitcoin/pull/20158) and linked PRs are an
```
Comment (jonatack) [inline _posts/2021-02-03-#20721.md:35]: for people with a slow internet connection, maybe append the PR name "De-globalize ChainstateManager"

File: _posts/2021-02-03-#20721.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    attempt to clarify and rationlize the interface to validation.
+
+- In early versions of Bitcoin, this distinction wasn't as clear. The [first
```
Comment (jonatack) [inline _posts/2021-02-03-#20721.md:38]: maybe s/Bitcoin/Bitcoin Core/

File: _posts/2021-02-03-#20721.md (hunk 2 above)
Comment (jonatack) [reply _posts/2021-02-03-#20721.md:35]: (same for the other PR links in the notes)

File: _posts/2021-02-03-#20721.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  9260](https://github.com/bitcoin/bitcoin/pull/9260) in 2016 that net processing
+  and validation were split into separate files.
+
+- Even today, the division between net, net processing and validation is not as
+  clear as we'd like. Clarifying and rationalizing these interfaces is an
+  ongoing project. See, for example, [PR
+  15141](https://github.com/bitcoin/bitcoin/pull/15141), which moves the denial
+  of service logic from validation into net processing, [PR
+  20158](https://github.com/bitcoin/bitcoin/pull/20158), which rationalizes the
+  interface to validation, and [PR
```
Comment (jonatack) [inline _posts/2021-02-03-#20721.md:51]: s/PR/issue/ ?

File: _posts/2021-02-03-#20721.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  19398](https://github.com/bitcoin/bitcoin/issues/19398), which moves
+  application-layer data and logic from net into net processing.
+
+- The high level goal of PR 19398 is to split responsibilities cleanly between
+  CConnman and PeerManager:
+
+  - C**Conn**man should be responsible for **connections** - accepting incoming
+    and opening outbound connections, opening and closing sockets, reading and
+    writing data on those sockets, and first-line-of-defence validation of
```
Comment (jonatack) [inline _posts/2021-02-03-#20721.md:60]: I think it's "defense" unless it's a UK spelling I'm unaware of

File: _posts/2021-02-03-#20721.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    deciding when to send messages to those peers.
+
+- Today's PR, [PR 20721](https://github.com/bitcoin/bitcoin/issues/20721), is
+  part of the larger project tracked by PR 19398. It moves the `ping` data
+  from net into net processing.
+
+## Notes
+
+- The `ping`/`pong` message exchange acts as a keepalive between peers on the
+  peer-to-peer network, and also measures round-trip latency. Best ping time
```
Comment (jonatack) [inline _posts/2021-02-03-#20721.md:74]: nit, omit comma

File: _posts/2021-02-03-#20721.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+  and most recent ping time are presented to users in the GUI and RPC interface,
```
Comment (jonatack) [inline _posts/2021-02-03-#20721.md:75]: GUI, RPC and CLI ;)

File: _posts/2021-02-03-#20721.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+  and best ping time is also used by the net layer in its peer eviction logic.
+
+- Some of the data members, `nPingNonceSent`, `m_ping_start` and `fPingQueued` are
+  only ever used by net processing, so are obvious candidates for moving into the
```
Comment (jonatack) [inline _posts/2021-02-03-#20721.md:79]: nit, s/so/so they/

File: _posts/2021-02-03-#20721.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+title: "Move ping data to net_processing"
+pr: 20721
+authors: [jnewbery]
+components: ["p2p"]
+host: jnewbery
+status: upcoming
+commit:
+---
+
+_Notes and questions to follow soon!_
```
Comment (jonatack) [inline _posts/2021-02-03-#20721.md:13]: remove this line ;)

**Review by jonatack — COMMENTED** [review]
Looks great apart from the minor comments.

Aside, one thing I was wondering is how to encourage participants to actually review rather than just showing up to discuss at the meeting. I wonder if more directions into the code (e.g. function names or commits) rather than links to click would help in encouraging to open the pull and codebase locally and dive into it a bit deeper. That may also reflect my personal preference to grep/look at code locally, which is faster and better (for me, maybe not everyone) than opening GitHub links that take a silly amount of time to load, moreso with a VPN or Tor. That may not be an issue for people who are in urban areas with fast internet or who don't use privacy tools as much (IDK). Just a thought.

File: _posts/2021-02-03-#20721.md (hunk 1 above)
Comment (jnewbery) [reply _posts/2021-02-03-#20721.md:36]: fixed!

File: _posts/2021-02-03-#20721.md (hunk 2 above)
Comment (jnewbery) [reply _posts/2021-02-03-#20721.md:35]: I'm not sure that giving the PR title "De-globalize ChainstateManager" is more illuminating than the description "and linked PRs are an attempt to clarify and rationalize the interface to validation".

File: _posts/2021-02-03-#20721.md (hunk 3 above)
Comment (jnewbery) [reply _posts/2021-02-03-#20721.md:38]: The software wasn't renamed "Bitcoin Core" until around 2013 if I recall correctly.

File: _posts/2021-02-03-#20721.md (hunk 4 above)
Comment (jnewbery) [reply _posts/2021-02-03-#20721.md:51]: Fixed

File: _posts/2021-02-03-#20721.md (hunk 5 above)
Comment (jnewbery) [reply _posts/2021-02-03-#20721.md:60]: Indeed it is my good chap: https://dictionary.cambridge.org/dictionary/english/defence :gb:

File: _posts/2021-02-03-#20721.md (hunk 6 above)
Comment (jnewbery) [reply _posts/2021-02-03-#20721.md:74]: done

File: _posts/2021-02-03-#20721.md (hunk 7 above)
Comment (jnewbery) [reply _posts/2021-02-03-#20721.md:75]: Hmmm, the CLI is a separate client which uses the RPC interface, it's not an interface itself.

File: _posts/2021-02-03-#20721.md (hunk 8 above)
Comment (jnewbery) [reply _posts/2021-02-03-#20721.md:79]: done

File: _posts/2021-02-03-#20721.md (hunk 9 above)
Comment (jnewbery) [reply _posts/2021-02-03-#20721.md:13]: :facepalm:

**Comment by jnewbery** [comment]
Thanks for the review @jonatack! Much appreciated, as always :)

> Aside, one thing I was wondering is how to encourage participants to actually review rather than just showing up to discuss at the meeting (which is great, maybe posting a review is asking a lot).

I'm not too worried about this. My original aim for the review club was to help people understand the process of Bitcoin Core. If people feel comfortable leaving a review after they've been to a few meetings, that's great.

One thing I want to be very careful to avoid is adding noise to the repo and making the maintainers' lives more difficult. If a lot of new contributors start leaving ACKs without any additional details, then that's very little additional information for maintainers to make their merge decision. This is a concern I've heard directly from maintainers.

File: _posts/2021-02-03-#20721.md (hunk 5 above)
Comment (jonatack) [reply _posts/2021-02-03-#20721.md:58]: > Indeed it is my good chap: https://dictionary.cambridge.org/dictionary/english/defence :gb: 

TIL--thanks! 🇬🇧

**Comment by jonatack** [comment]
That makes sense 👍

**Pull request merged**